OpenSim/Common/Component.h

```diff
@@ -2117,12 +2117,15 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(Component, Object);
     // End of System Creation and Access Methods.
     //@} 
 
-
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wunsupported-friend"
+#if defined(__clang__)
+    #pragma clang diagnostic push
+    #pragma clang diagnostic ignored "-Wunsupported-friend"
+#endif
```

Comment:
Omitting the `#if defined (__clang__)` caused a `unknown pragma` warning with MSVC.